activestorage/app/models/active_storage/blob.rb

```diff
@@ -168,6 +168,14 @@ def filename
     ActiveStorage::Filename.new(self[:filename])
   end
 
+  def custom_metadata
+   self[:metadata][:custom] || {}
+  end
+
+  def custom_metadata=(metadata)
+    self[:metadata].merge!(custom: metadata)
```

Comment:
```suggestion
    self[:metadata] = self[:metadata].merge(custom: metadata)
```

Relying in the mutation of the object to save in the database might bring us problems in the future. Better to create a new object.